unittest/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+
+include(ExternalProject)
```

Comment:
@junghans I would appreciate, if you could have a look at what I hacked together here. It is based on what you have done in the LAMMPS cmake tree and suggestions gathered from multiple sources on the web (some of them for CMake versions older than 2.8). The simplest option would require CMake 3.11, but that we cannot for at least 2 more years until we can declare ubuntu 18.04 as only partially supported (same as we currently do with ubuntu 16.04).
It appears to be working with GNU gcc, and Clang using GNU make and ninja. Have not tried MacOSX yet...